### ISSUE TITLE
glfw3native.h undefines APIENTRY from windows.h

### ISSUE THREAD

**Issue by lcomstive** [glfw/glfw#1062]
Defining `GLFW_EXPOSE_NATIVE_WIN32` then including `glfw3native.h` produced errors when attempting to use [glDebugMessageCallback](https://www.khronos.org/opengl/wiki/Debug_Output) . To fix it I simply commented out `#undef APIENTRY` on line 89 of glfw3native and everything was working as it should. Is this just me?

OS: Windows 10 x64
Compiler: Visual Studio 15 2017
GLFW Version: commit af866e05d20068be149e0cc410d64490fb0b7a8c (also tried with release 3.1.2)
GLAD Version: 3.3 Compatibility (no extensions)(generated via [official site](http://glad.dav1d.de/))

```cpp
// Code to reproduce
#ifdef _WIN32
#define GLFW_EXPOSE_NATIVE_WIN32
#endif
#include <glad/glad.h>
#include <GLFW/glfw3.h>
#include <GLFW/glfw3native.h>

// not specifying 'APIENTRY' causes access violation errors
// specifiying 'APIENTRY' causes compiler errors such as 'inconsistent return statement', 
//     'illegal use of type void' and 'missing type specifier - int assumed'
void APIENTRY DebugCallback(GLenum source, GLenum type, 
                                              GLuint id, GLenum severity, GLsizei length, 
                                              const GLchar* message, const void* userObject) 
{ cout << "[OPENGL_DEBUG] " << message << endl; }

int main(int argc, char** argv)
{
    if(!glfwInit())
        return -1;
    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
    glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_COMPAT_PROFILE);
    glfwWindowHint(GLFW_OPENGL_DEBUG_CONTEXT, TRUE);

    GLFWwindow* window = glfwCreateWindow(800, 600, "OpenGL", NULL, NULL);
    glfwMakeContextCurrent(window);
    gladLoadGLLoad((GLADloadproc)glfwGetProcAddress);
    if(!gladLoadGL())
        return -2;

    if(GL_VERSION_4_3)
    {
        glEnable(GL_DEBUG_OUTPUT);
        glEnable(GL_DEBUG_OUTPUT_SYNCHRONOUS);
        glDebugMessageCallback(DebugCallback, NULL);
    }

    while(!glfwWindowShouldClose(window))
    {
        glfwPollEvents();
        glfwSwapBuffers(window);
    }
    glfwDestroyWindow(window);
    glfwTerminate();
}
```

**Comment by elmindreda** [comment]
> Is this just me?

Nope, that's a real bug. Thank you for reporting it!

It's happening because `glfw3native.h` assumes that the `APIENTRY` it is undefining comes from `glfw3.h` but in this case it comes from `windows.h` as included by `glad.h`. When `windows.h` is then re-included nothing happens and it remains undefined.

**Comment by elmindreda** [comment]
A quick fix is to change the order a little.

```c
#define GLFW_INCLUDE_NONE
#ifdef _WIN32
#define GLFW_EXPOSE_NATIVE_WIN32
#endif
#include <GLFW/glfw3.h>
#include <GLFW/glfw3native.h>
#include <glad/glad.h>
```

Will make a proper fix in the meantime.

**Comment by elmindreda** [comment]
Some kind of fix has been made with c23fca63439f2456f5aa0b99a724bbd7177e0167.
